lib/install/management/appliance.go

```diff
@@ -120,25 +121,50 @@ func (d *Dispatcher) checkExistence(conf *config.VirtualContainerHostConfigSpec,
 		return err
 	}
 	if orp == nil {
-		return nil
+		return nil // TODO[zjs]: Check for existing VM Group in this case too
 	}
 
 	rp := compute.NewResourcePool(d.op, d.session, orp.Reference())
 	vm, err := rp.GetChildVM(d.op, d.session, conf.Name)
 	if err != nil {
 		return err
 	}
-	if vm == nil {
-		return nil
+	if vm != nil {
+		d.op.Debug("Appliance is found")
+		if ok, verr := d.isVCH(vm); !ok {
+			verr = errors.Errorf("Found virtual machine %q, but it is not a VCH. Please choose a different virtual app.", conf.Name)
```

Comment:
Some words that aren't `virtual app` in the error message.
`Specified virtual machine %q is not a VCH endpointVM.`

Reply:
- [ ] Clean up wording in log messages in `lib/install/management/appliance.go`.